nilearn/reporting/_get_clusters_table.py

```diff
@@ -186,10 +260,7 @@ def get_clusters_table(stat_img, stat_threshold, cluster_threshold=None,
                               copy_data=(cluster_threshold is not None))
 
     # Define array for 6-connectivity, aka NN1 or "faces"
-    conn_mat = np.zeros((3, 3, 3), int)
-    conn_mat[1, 1, :] = 1
-    conn_mat[1, :, 1] = 1
-    conn_mat[:, 1, 1] = 1
+    conn_mat = ndimage.generate_binary_structure(rank=3, connectivity=2)
```

Comment:
I might be wrong, but I don't think this is equivalent to the previous code:

```python
>>> ndimage.generate_binary_structure(rank=3, connectivity=2).astype(int)
array([[[0, 1, 0],
        [1, 1, 1],
        [0, 1, 0]],

       [[1, 1, 1],
        [1, 1, 1],
        [1, 1, 1]],

       [[0, 1, 0],
        [1, 1, 1],
        [0, 1, 0]]])
```

Reply:
You're right, thank you for catching that. I copied this from another repo where I'm using NN2 instead of NN1.

Reply:
```suggestion
    conn_mat = ndimage.generate_binary_structure(rank=3, connectivity=1)
```

Reply:
```python
import numpy as np
from scipy import ndimage

conn_mat = np.zeros((3, 3, 3), int)
conn_mat[1, 1, :] = 1
conn_mat[1, :, 1] = 1
conn_mat[:, 1, 1] = 1
conn_mat = conn_mat.astype(bool)

conn_mat_new_bad = ndimage.generate_binary_structure(rank=3, connectivity=2)
print(np.array_equal(conn_mat, conn_mat_new_bad))
# False
conn_mat_new = ndimage.generate_binary_structure(rank=3, connectivity=1)
print(np.array_equal(conn_mat, conn_mat_new))
# True
```